File: MyWallet/Backend_MyWallet/src/main/java/org/koz/wallet/app/transaction/model/TransactionCategory.java
package org.koz.wallet.app.transaction.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TransactionCategory {
    private Long id;
    private Long userId;
    private String name;
    private String keywords;
}

File: MyWallet/Front_MyWallet/src/app/_model/transaction_category.ts
export class TransactionCategory {
  id: number;
  name: string;
  keywords: string;
  default: boolean;

  public static clone(transactionCategory: TransactionCategory): TransactionCategory {
    return {...transactionCategory, default: false};
  }

  public static of(name: string, keywords: string): TransactionCategory {
    let transactionCategory: TransactionCategory = new TransactionCategory();
    transactionCategory.name = name;
    transactionCategory.keywords = keywords;
    transactionCategory.default = false;
    return transactionCategory;
  }
}
